atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableDropper.java

```diff
@@ -39,68 +38,68 @@
+            CassandraTableTruncator cassandraTableTruncator) {
         this.config = config;
         this.clientPool = clientPool;
         this.cellValuePutter = cellValuePutter;
         this.wrappingQueryRunner = wrappingQueryRunner;
-        this.deleteConsistency = deleteConsistency;
+        this.cassandraTableTruncator = cassandraTableTruncator;
     }
 
     void dropTables(final Set<TableReference> tablesToDrop) {
         try {
-            clientPool.runWithRetry(
-                    (FunctionCheckedException<CassandraClient, Void, Exception>) client -> {
-                        KsDef ks = client.describe_keyspace(
-                                config.getKeyspaceOrThrow());
-                        Set<TableReference> existingTables = Sets.newHashSet();
+            clientPool.runWithRetry(client -> {
+                KsDef ks = client.describe_keyspace(config.getKeyspaceOrThrow());
+                Set<TableReference> existingTables = Sets.newHashSet();
 
-                        existingTables.addAll(ks.getCf_defs().stream()
-                                .map(CassandraKeyValueServices::tableReferenceFromCfDef)
-                                .collect(Collectors.toList()));
+                existingTables.addAll(ks.getCf_defs().stream()
+                        .map(CassandraKeyValueServices::tableReferenceFromCfDef)
+                        .collect(Collectors.toList()));
 
-                        for (TableReference table : tablesToDrop) {
-                            CassandraVerifier.sanityCheckTableName(table);
-                            if (existingTables.contains(table)) {
-                                client.system_drop_column_family(
-                                        CassandraKeyValueServiceImpl.internalTableName(table));
-                                putMetadataWithoutChangingSettings(table,
-                                        AtlasDbConstants.EMPTY_TABLE_METADATA);
-                            } else {
-                                log.warn("Ignored call to drop a table ({}) that did not exist.",
-                                        LoggingArgs.tableRef(table));
-                            }
-                        }
-                        CassandraKeyValueServices.waitForSchemaVersions(config, client, "after dropping the column "
-                                + "family for tables " + tablesToDrop + " in a call to drop tables");
-                        return null;
-                    });
+                for (TableReference table : tablesToDrop) {
+                    CassandraVerifier.sanityCheckTableName(table);
+                    if (existingTables.contains(table)) {
+                        cassandraTableTruncator.runTruncateOnClient(ImmutableSet.of(table), client);
+                        client.system_drop_column_family(CassandraKeyValueServiceImpl.internalTableName(table));
+                        updateMetadataForTable(table);
```

Comment:
What is this metadata used for? Is it correct to update the metadata after truncating and deleting the table?

Reply:
Bad naming, renamed to make clear

Reply:
To elaborate, we don't use this metadata to determine if a table exists for future deletions and creations, so having some potentially stale metadata is fine.